src/index.ts

```diff
@@ -115,15 +115,26 @@ export function project<T = any>(options?: string | Options) {
   return metadataAccessor<T>('project', options);
 }
 
+/*
+ * How many times should we retry detecting GCP environment.
+ */
+function detectGCPAvailableRetries(): number {
+  return process.env.DETECT_GCP_RETRIES
+    ? Number(process.env.DETECT_GCP_RETRIES)
```

Comment:
It's ok if this ends up as a NaN right? Seems like it might do the same thing as zero anyway

Reply:
I think this edge-case is okay, I mostly want this option here to allow for some directed experiments with customers. If we find that a certain number of retries works better, we might remove this configuration and try to bake retries back into the client.